sled-agent/src/bootstrap/agent.rs

```diff
@@ -827,28 +845,39 @@ impl Agent {
         &self,
         _state: &tokio::sync::MutexGuard<'_, SledAgentState>,
     ) -> Result<(), BootstrapError> {
-        tokio::fs::remove_dir_all(omicron_common::OMICRON_CONFIG_PATH)
+        let config_dirs = self
+            .storage_resources
+            .all_m2_mountpoints(sled_hardware::disk::CONFIG_DATASET)
```

Comment:
This is *basically* the same code it was before, but we're removing everything within the dataset, rather than deleting the dataset itself.